src/darker/tests/test_main.py

```diff
@@ -261,14 +267,22 @@ def test_main(
     expect_stdout,
     expect_a_py,
     expect_retval,
+    root_as_cwd,
+    tmp_path_factory,
 ):  # pylint: disable=too-many-arguments
     """Main function outputs diffs and modifies files correctly"""
-    monkeypatch.chdir(git_repo.root)
+    if root_as_cwd:
+        cwd = git_repo.root
+        pwd = ""
+    else:
+        cwd = tmp_path_factory.mktemp("not_a_git_repo")
+        pwd = str(git_repo.root) + "/"
```

Comment:
In Darker I've tried to consistently concatenate path elements using `pathlib.Path` methods instead of string operations. So here, too we could do instead:
```suggestion
    if root_as_cwd:
        cwd = git_repo.root
        pwd = Path("")
    else:
        cwd = tmp_path_factory.mktemp("not_a_git_repo")
        pwd = git_repo.root
```

A corresponding change is needed below for line 285.